Birthday-2024-Project/Scripts/ScreenManagement/ScreenLogic.gd

```diff
@@ -0,0 +1,37 @@
+class_name ScreenLogic
+extends Node
+
+signal ScreenEnter
+signal ScreenExit
+
+@export var clearScreenStack : bool = true
+
+var screenManager : ScreenManager
+var transitionData : Dictionary
+
+func ClosePopup(data : Dictionary):
+	screenManager.CloseTopScreen(data)
+	
+func ClosePopupBySignal():
+	screenManager.CloseTopScreen({})
+
+func GoToScreen(screen : PackedScene, data : Dictionary):
+	screenManager.GoToScreen(screen, data)
+
+func ShowSettings():
+	screenManager.ShowSettings()
+
+func ShowConfirmationPopup(title : String, body : String, confirm : String = "OK", cancel : String = "Cancel"):
+	screenManager.ShowConfirmationPopup(title, body, confirm, cancel)
+
+func ExitApplication():
+	ScreenEnter.connect(ExitConfirmation)
+	screenManager.ShowConfirmationPopup("Quit?", "Are you sure you want to quit?", "Yes", "No")
```

Comment:
instead of calling screenManager.ShowConfirmationPopup() it should call for ShowConfirmationPopup() from this ile from line 24

Reply:
Now using the local function version